### ISSUE TITLE
LUA error in 1st wipe

### ISSUE THREAD

**Issue by Krovikan** [DeadlyBossMods/DeadlyBossMods#72]
I expect message of wipe, but instead I get message of wipe and a LUA error about Playsond in DBM-Core. Only success in the 1st wipe of all session of the game.

Error:

Date: 2019-09-14 16:54:52
ID: 1
Error occured in: Global
Count: 1
Message: ..\AddOns\DBM-Core\DBM-Core.lua line 6418:
   PlaySoundFile Error - Invalid fileDataID for sound.
Debug:
   [C]: ?
   [C]: ?
   DBM-Core\DBM-Core.lua:6418:
      DBM-Core\DBM-Core.lua:6394
   DBM-Core\DBM-Core.lua:6428: PlaySoundFile()
   DBM-Core\DBM-Core.lua:6028: EndCombat()
   DBM-Core\DBM-Core.lua:5584: func()
   DBM-Core\DBM-Core.lua:1700:
      DBM-Core\DBM-Core.lua:1691
Locals:

AddOns:
  Swatter, v7.3.1 (<%codename%>)
  ACP, v3.5.6-2-g9b55b72 
  AdvancedInterfaceOptions, v1.3.2-1-g9719ccc
  Altoholic, vr191
  AtlasLoot, vv8.10.00
  AtlasLootBattleforAzeroth, vv8.10.00
  AtlasLootBurningCrusade, vv8.10.00
  AtlasLootCataclysm, vv8.10.00
  AtlasLootClassic, vv8.10.00
  AtlasLootLegion, vv8.10.00
  AtlasLootMistsofPandaria, vv8.10.00
  AtlasLootWarlordsofDraenor, vv8.10.00
  AtlasLootWrathoftheLichKing, vv8.10.00
  BagBrother, v
  Bagnon, v8.2.9
  Bartender4, v4.8.7-2-gb6977ae
  BasicThreatBar, vr95-beta
  ChampionCommander, v1.2.2 80200
  DataStore, vr66
  DataStoreAgenda, vr30
  DataStoreAuctions, vr51
  DataStoreCharacters, vr44
  DataStoreContainers, vr60
  DataStoreCrafts, vr83
  DataStoreCurrencies, vr39
  DataStoreInventory, vr54
  DataStoreMails, vr53
  DataStoreSpells, vr33
  DataStoreStats, vr32
  DataStoreTalents, vr62
  DBMArchaeology, vr33-3-gc2341af
  DBMAzerothBfA, v
  DBMCore, v8.2.18-10-g05d5954
  DBMDefaultSkin, v
  DBMGarrisonInvasions, vr44-4-g98e9fb6
  DBMIcecrown, vr289-17-g9c7e74e
  DBMStatusBarTimers, v
  Decursive, v2.7.6.4
  ExRT, v4030
  GatherMate2, v1.45
  GatherMateSharing, vv1.12
  HandyNotes, vv1.5.6
  HandyNotesBattleForAzerothAchievments, vv7
  HandyNotesBattleForAzerothTreasures, v1
  HandyNotesCamelFigurines, v1.3
  HandyNotesHallowsEnd, vv1.4.3-8-ga38054e
  HandyNotesLunarFestival, vv1.6.2
  HandyNotesSummerFestival, vv1.6.1-2-gd05b89c
  InstanceAchievementTracker, v2.57.0
  LibSharedMedia30, v3.0-114
  MapCoords, v0.31
  MountJournalEnhanced, v2.5.1
  NeatPlates, v8.2.0.14
  NeatPlatesNeon, v
  NeatPlatesHub, v
  NeatPlatesWidgets, v
  Nomepidas, v
  Overachiever, v1.0.7
  OverachieverCache, v1.0.7
  OverachieverTabs, v1.0.7
  OverachieverTrade, v1.0.7
  Paste, v1.4.10
  PetBattleTeams, v3.3.11
  PetJournalEnhanced, v2.9.30
  PetTracker, v8.2.1
  PetTrackerBreeds, v8.2.1
  PetTrackerJournal, v8.2.1
  PetTrackerUpgrades, v8.2.1
  Quartz, v3.5.1-1-gac72c9a
  Skada, v1.7.6
  SlideBar, v7.3.1 (<%codename%>)
  TomTom, vv80200-1.0.7
  ToyBoxEnhanced, v2.3
  WarfrontRareTracker, v1.0.7
  WeakAuras, v2.14.6-33-g8a60d77
  WIM, v3.7.33-3-g7e32919
  WorldQuestTracker, v
  ZPerl, v5.7.8
  ZPerlArcaneBar, v
  ZPerlParty, v
  ZPerlPlayer, v
  ZPerlPlayerBuffs, v
  ZPerlTarget, v
  ZPerlTargetTarget, v
  BlizRuntimeLib_esES v8.2.0.80200 <none>
  (ck=913)

To reproduce:

1. Go to any boss of the raid.
2. Wipe all.
3. See LUA error in the screen (I use Swatter to catch errors)

The moment of the LUA error (September 10):

https://i.imgur.com/WFJ89T9.png

I don't try to only enable DBM.

Current version: 8.2.18-10-g05d5954

Was working always but I don't remember from wich version fail. I think fail before September.

From years ago I check always everyday for the last version (alpha) of the addon in Twitch/Curse.

The night of the screenshot I was died in Behemoth but no Error. Only success the Error in a General Wipe (all the raid).

Regards

**Comment by Developaws** [comment]
What is your "Set sound played for encounter wipe" set to? :)

**Comment by Krovikan** [comment]
Sound event for wipe is Yogg-Saron: Laugh.

**Comment by MysticalOS** [comment]
Checked that sound and it's fine, plays correctly, has valid soundID in config. Since you are using alpha, you can use debug I added to figure out what's ACTUALLY causing these errors for users. do /dbm debug

then next time it happens tell us what chat says is the file it tried to play.

**Comment by Krovikan** [comment]
Ok.

I need to launch these 2 commands (one after another) at the start of the raid and I have upload this file at the end of raid (2 hours)?

The file will be generated in Logs directory of WoW?
 
Regards

**Comment by MysticalOS** [comment]
nothing to upload, you'll see the print in chat log.

**Comment by Krovikan** [comment]
Thanks for the clarification. I will try tomorrow.

Regards

**Comment by Krovikan** [comment]
Is strange but today, I enable /dbm debug on raid and I don't have spam in chat (I use /dbm debuglevel 3 to stop debugger after the 1st Boss encounter). And no error.

The sound Yogg-Saron: Laugh was listened at wipe.

ENCOUNTER_END event fired:2304 Lady Gobernalle 15 12 0

I will continue monitorizing if the problem reproduce. But for now, all OK.


Regards

**Comment by MysticalOS** [comment]
Debuglevel 3 doesn't stop it, that should hae been run IMMEDIATELY after turning debug mode on, that tells it what level of debug to do. doing /dbm debug a second timeis what turns it off.

If sound played correctly after wipe you probably already fixed it by clicking drop down causing it to set the value over whatever bad value was there before.

**Comment by Krovikan** [comment]
Yes, when @developaws ask me for the sound, I cliked the sound to listen.

**Comment by MysticalOS** [comment]
Ah, that would have repaired the issue right then and there, because I know what issue was, whatever sound it was ACTUALLY set to was missing/removed, when a sound is missing the menu may show a sound but that doesn't mean THAT is what it is set to. It just means table key is missing so menu is showing something else. But at time you were getting errors it wasn't ACTUALLy set to yogg laugh, that's just what menu said cause what it was set to was missing. Of course when you clicked it, you actually set it to yogg laugh.

TL/DR, no point in running debug now, we'll never know what sound you had before that ended up missing.
